src/Hl7.Fhir.Core/Rest/FhirClient.cs

```diff
@@ -1058,9 +1058,11 @@ protected virtual void AfterResponse(HttpWebResponse webResponse, byte[] body)
                 if (result is OperationOutcome)
                     return null;
 
-                var message = String.Format("Operation {0} on {1} expected a body of type {2} but a {3} was returned", response.Request.Method,
-                    response.Request.Url, typeof(TResource).Name, result.GetType().Name);
-                throw new FhirOperationException(message, _requester.LastResponse.StatusCode);
+                var message = String.Format("Operation {0} on {1} expected a body of type {2} but a {3} was returned", request.Request.Method,
+                    request.Request.Url, typeof(TResource).Name, result.GetType().Name);
+
+                Enum.TryParse(response.Response.Status, out HttpStatusCode code);
```

Comment:
Why do you switch to use response.Response.Status instead of _requester.LastResponse.StatusCode?